### ISSUE TITLE
Alarms and Sirens Update + Voluntary Sound Contributions and Modeling

### ISSUE THREAD

**Issue by genestrashcan** [age-series/ElectricalAge#189]
@jrddunbr - 

In a future version lets rework the sirens and alarms, and make new models of them. I will probably do some modeling and produce the sounds for you in the future sometime hopefully.

For sirens we would model them to look like actual sirens you would see on top of a fire department etc.

for sirens i was thinking 3 distinct types to make models of and sounds

- Standard Federal Signal Siren
![710d6ef3-5f4e-4dd4-9be9-1419b7f42e40_d](https://user-images.githubusercontent.com/11165643/66064381-44079e00-e513-11e9-8a6e-03cf8a704a76.jpg)

Sound of a Good Standard Siren - https://www.youtube.com/watch?v=Tw3L1iteduo

- Diaphone also known as the Horn from Hell or a Mad Duck Horn!
![24930377caca463b6470a3818c672900](https://user-images.githubusercontent.com/11165643/66064350-36521880-e513-11e9-886c-0cc8c1ddcb1a.jpg)

Sound of a good diaphone to use for the Diaphone (use the early 2 tone honks in the video before it changes to single tone around 0:21) - https://www.youtube.com/watch?v=Dluep2RYV7s

- Ktg 9/10 Also known as a Hoopla Siren
![kockum-sonics-ktg-uztenstorf 33](https://user-images.githubusercontent.com/11165643/66064455-68637a80-e513-11e9-9afd-5994bb6b6523.jpg)

Good Sound of a Ktg - https://www.youtube.com/watch?v=bI6hI2WEopM

for alarms we would model them after the alarm you may see in a school used for a fire drill.

simplex 4040 

![23735074860_0e5f2681a0_b](https://user-images.githubusercontent.com/11165643/66065060-a319e280-e514-11e9-8587-6bb53535d534.jpg)

Sound for the Simplex - https://www.youtube.com/watch?v=uayunq6YEho (Use March Time)

**Comment by jrddunbr** [comment]
Sirens are a cool thing. Just make sure that we either own the rights to the sounds, or that they are compatibly licensed with the project.

I'll see the models as they come. Upload them here or on the Discor (or do a PR?)

**Comment by mario2582** [comment]
the sounds will be public domain

**Comment by mario2582** [comment]
Can't wait to see some players get a jumpscare when you get servers with EA working and someone builds a powerplant and puts a gamewell diaphone on top of it and someone else is on the server near the powerplant and it suddenly goes off and starts honking.

**Comment by jrddunbr** [comment]
Heh, that's definitely not how fair use works. Ask for permission from the creators.

**Comment by jrddunbr** [comment]
This particular video you linked is CC-BY which we can use: https://www.youtube.com/watch?v=Dluep2RYV7s

**Comment by mario2582** [comment]
Here's a good one for the regular siren with a CC-BY Licence.

https://www.youtube.com/watch?v=HPixqLqy3Hg

now to try and find one for the Ktg Siren

**Comment by Ye-Old-Apple** [comment]
Siren Models, Including a STH-10, 3T22, KTG-10, Diaphone Horn, 4040 Fire alarm
[Sirens.zip](https://github.com/jrddunbr/ElectricalAge/files/3934955/Sirens.zip)
I also quickly cut and edited the sound for the Diaphone Horn with the two tone honks.

**Comment by mario2582** [comment]
@Ye-Old-Apple - The exit mouth on the horn on your Gamewell diaphone horn blender model is too wide... Images below showing what model and a proper Gamewell diaphone horn looks like...

![bulletin5010_1_print - Copy](https://user-images.githubusercontent.com/11165643/71221108-40292380-2299-11ea-81b1-4db19f4d404f.jpg)

![bulletin5010_1_print - Copy (2)](https://user-images.githubusercontent.com/11165643/71221121-53d48a00-2299-11ea-9e66-aab8a6b75b7e.jpg)

![bulletin5010_1_print](https://user-images.githubusercontent.com/11165643/71221125-559e4d80-2299-11ea-9112-e4a95a88af48.jpg)

**Comment by mario2582** [comment]
@Ye-Old-Apple - Diaphone Horn Blend Model with Narrower horn

[Gamewell Diaphone Horn Siren.zip](https://github.com/jrddunbr/ElectricalAge/files/3986227/Gamewell.Diaphone.Horn.Siren.zip)

**Comment by Ye-Old-Apple** [comment]
ok

**Comment by mario2582** [comment]
@Ye-Old-Apple @jrddunbr - Here are the close and distant sounds for the Ktg / Hoopla Siren, I used a sound generating tool to recreate a highly accurate mimic of the ktg siren sound as accurately as I could my self and dedicated them to the public domain. 

[Ktg Siren Sounds.zip](https://github.com/jrddunbr/ElectricalAge/files/3992107/Ktg.Siren.Sounds.zip)

**Comment by mario2582** [comment]
@Ye-Old-Apple @jrddunbr - Here are the close and distant sounds for the Ktg / Hoopla Siren, I recorded the siren sound myself and had the sound dedicated to the public domain. The distant is a highly edited version of the original in audacity to make it sound far away. Also attached is a distant sound for the gamewell diaphone too...

[Ktg Siren Sounds.zip](https://github.com/jrddunbr/ElectricalAge/files/3992108/Ktg.Siren.Sounds.zip)

[Diaphone_Horn_distant.zip](https://github.com/jrddunbr/ElectricalAge/files/3992109/Diaphone_Horn_distant.zip)

**Comment by mario2582** [comment]
@jrddunbr - For the Gamewell diaphone, the mechanical clockwork disc system they used to control the honks seen in these videos should be implemented and modeled too. In the second video, the device you saw in the first one (inside the giant red box the bell is wired into) would also honk the Gamewell horn every time the bell rang.

the device you design in-game to send a signal to honk the diaphone should have several mechanical discs that control the honk pattern that can be selected from like the one in real life for different variants of honks when it is set off. (in real life they used different discs for different emergency codes the horn would honk.)

if the honk control mechanism is modeled, I will provide sounds for close and distant versions of the shorter honk that the mechanical disc would make playback for each honk of the horn.

[](https://www.youtube.com/watch?v=jaeYl09lU-k)

[](https://www.youtube.com/watch?v=3Bq7pUQWuFU)

**Comment by jrddunbr** [comment]
So loading different models of a disk inside is certainly possible, but it is difficult to sync sounds with rotating objects positions. I have never seen an alarm like that.

**Comment by mario2582** [comment]
@jrddunbr - to make the discs work all you would have to do is mimic the device in the video and make it so the disc would make the device send a signals in a repeating pattern to the diaphone which would make it honk. I will make a shorter version of the honk just for this purpose.

**Comment by jrddunbr** [comment]
The code is a little different than just telling it to move when the arm moves due to the way the network code works, but we could figure that out.

**Comment by mario2582** [comment]
To make it work all we have to do is code the diaphone siren to only play the short one honk when it receives signal voltage through a signal wire, if the voltage is a constant near max voltage for a signal wire in the signal wire used, the diaphone will only honk once and no longer honk until the voltage has returned to something near 0 and when the voltage was pulsed near-maximum again, the diaphone would honk again. then all you have to do is make the device that would use the rotary discs to push a button in a pattern you would want that would send a short 50v signal down the line connected to the diaphone every time the rotary disc pressed the button that would generate a short voltage pulse on the signal cable to honk the horn.

**Comment by mario2582** [comment]
@jrddunbr  Here is the sounds for the shorter Pulsed Diaphone

[Diaphone Pulse Sound.zip](https://github.com/jrddunbr/ElectricalAge/files/3992927/Diaphone.Pulse.Sound.zip)

**Comment by Ye-Old-Apple** [comment]
this sounds a tad over-complicated for a single horn. I thought the only point of their being different horns is for the different sounds.

**Comment by Grissess** [comment]
Indeed, at least transitionally, the "Standard Alarm" can be equipped with a choice, _a la_ IC2. At some later point, the models can be differentiated with shared code (like the lamps).

I can probably take the "transitional" part after the holidays, though I will say: GUI is annoying.

**Comment by jrddunbr** [comment]
This issue is being closed now since there will be no more feature updates in Electrical Age for Minecraft 1.7.10.
If the issue is relevant to 1.12, it will be re-opened there at a later date.

**Comment by mario2582** [comment]
> This issue is being closed now since there will be no more feature updates in Electrical Age for Minecraft 1.7.10.
> If the issue is relevant to 1.12, it will be re-opened there at a later date.

@jrddunbr - Please reopen this + others and label them for Future version development or something.

Closing out all the issues / dumping all the older code / assets that contain WIP code and assets destined for a future version in a program is very bad software development practice.

**Comment by jrddunbr** [comment]
Don't tell me how to operate an open source community.

I mentioned that we will be re-opening these issues on the new repository. Such issues already have a label for being reopened later (`1.12`).
